tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -6466,87 +6466,68 @@ NodeDef GetSquaredDifferenceNodeDef(DataType dtype) {
   return squared_diff.operation.node()->def();
 }
 
-template <DataType dtype>
-void TestConvertSquaredDifference(OpConverterTest* test) {
-  typedef typename EnumToDataType<dtype>::Type CType;
+TEST_P(OpConverterTest2, ConvertSquaredDifference) {
+  {
+    // Input is a weight, should fail.
+    Reset();
+    NodeDef node_def = GetSquaredDifferenceNodeDef(tf_dtype);
+    AddTestWeights<float>("x", {1, 2, 3}, {1, 2, 3, 4, 5, 6});
+    AddTestTensor("y", {1, 1, 2, 3});
+    RunValidationAndConversion(node_def, error::UNIMPLEMENTED,
+                               "The input \"x\" for SquaredDifference must be "
+                               "a tensor, at my_squared_diff");
+  }
 
   struct TestParams {
     std::vector<int> dims_x;
     std::vector<int> dims_y;
-    std::vector<CType> value_x;
-    std::vector<CType> value_y;
+    std::vector<float> value_x;
+    std::vector<float> value_y;
     std::vector<int> expected_output_dims;
-    std::vector<CType> expected_output;
+    std::vector<float> expected_output;
+    Status status;
+    Status runtime_status;
   };
 
-  const std::vector<CType> common_input = InitTestVector<CType>(6);
+  const std::vector<float> common_input = InitTestVector<float>(6);
   std::vector<TestParams> params = {
+      {/*dims_x=*/{1, 2, 3},
+       /*dims_y=*/{1, 7, 5},
+       /*value_x=*/common_input,
+       /*value_y=*/std::vector<float>(7 * 5, 0),
+       /*expected_output_dims=*/{1, 1, 2, 3},
+       /*expected_output=*/common_input,
+       trt_mode == TrtTestMode::kDynamicShape
+           ? Status::OK()
+           : Status(error::INVALID_ARGUMENT, "Infeasible broadcast scheme"),
+       Status(error::INVALID_ARGUMENT, "Negative number of dimensions -1")},
```

Comment:
Do you know why we delay the handling of error until [here](https://github.com/tensorflow/tensorflow/blob/a26381f3dc49cfe7ef4bdc05652fc71b62f932f1/tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc#L43) returns dims.nbDims == -1? 
I see that there is an error in building the engine before that:
 trt_logger.cc:42] DefaultLogger (Unnamed Layer* 5) [Shuffle]: reshaping failed for tensor: input

Reply:
Maybe we should not pass nbDims==-1 to [here](https://github.com/tensorflow/tensorflow/blob/a26381f3dc49cfe7ef4bdc05652fc71b62f932f1/tensorflow/compiler/tf2tensorrt/utils/trt_engine_utils.cc#L54).  Do you think we should return an error before calling TrtDimsToTensorShape, that says either binding_index is out of bound or the input shapes do not lead to a valid TRT engine, and put a comment  explaining TRT current doesn't report an error in engine build time?

Reply:
That is a good idea, I have implemented it.